### PR TITLE
new global_nest_v1 suite and #1965

### LINKED ISSUE #1964: Updates to C3 and SAS cumulus convection schemes
## Description
Optimal tuning parameters are found to be different in the prognostic closure formulation shared by SAS and C3 convection schemes for RRFS/GFS and HAFS applications. Therefore I propose to bring them out as namelist parameters.

## Solution
Create a PR that brings out the parameters as tuning parameters.

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
## PR Author Checklist:

- [x] I have linked PR's from all sub-components involved in section below. 
- [x] I am confirming reviews are completed in ALL sub-component PR's.
- [x] I have run the full RT suite on either Hera/Cheyenne AND have attached the log to this PR below this line:
  - LOG: Committed to branch. Tested on hera and jet.
- [x] I have added the list of all failed regression tests to "Anticipated changes" section.
- [x] I have filled out all sections of the template.

## Description
This is a suite of physics schemes nominally planned for the RRFS v2 upgrade. It's being used for the Water in the West project to simulate atmospheric rivers using a global model with a static nest. Hence "global_nest_v1."

Includes #1965 which fixes issue #1964

### Associated UFSWM Issue to close

- fixes https://github.com/ufs-community/ufs-weather-model/issues/1964
- closes #1965 

### Subcomponent Pull Requests
- FV3: https://github.com/NOAA-EMC/fv3atm/pull/709
- ccpp-physics: https://github.com/ufs-community/ccpp-physics/pull/122
- Lisa ccpp-physics: https://github.com/lisa-bengtsson/ccpp-physics/pull/3

### Blocking Dependencies
None.

### Subcomponents involved:
- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [x] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [ ] none

### Input data
- [x] No changes are expected to input data.

### Regression Tests:
- [x] No changes are expected to any regression test.
- [x] Changes are expected to the following tests:

There are two new tests:

1. gnv1_c96_no_nest_debug - The standard C96 initial conditions used for most physics suites, configured to run global_nest_v1. This is run in debug mode for both compilers
2. gnv1_nested - Run off of the HAFS global static nest test initial conditions, this tests the suite in a static nest.

### Libraries

- [x] Not Needed

<details><summary>Code Managers Log</summary>

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR.
- [ ] Move new/updated input data on RDHPCS Hera and propagate input data changes to all supported systems.
  - [ ] N/A

### Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [ ] Cheyenne
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment
</details>

### CODE REVIEW

**Comment by SamuelTrahanNOAA** [comment]
I have already tested this on top of #1942, yesterday. I'm testing on top of #1953 now.

**Comment by SamuelTrahanNOAA** [comment]
It works on top of #1953 too. I have not pushed the changes yet because I don't own the ccpp-physics branch. After #1953 is merged, I'll ask Lisa to update her ccpp-physics branch, and I'll push my changes.

**Comment by SamuelTrahanNOAA** [comment]
I'll be gone from now until about 1PM Eastern, 11 AM Mountain.

**Comment by SamuelTrahanNOAA** [comment]
All components are up to date. This branch is ready for testing.

You will need to generate baselines for the gnv1_nested and gnv_c96_no_nest_debug tests. No tests will change results.

**Comment by SamuelTrahanNOAA** [comment]
It is important that you do *not* generate new baselines for all tests. Just generate baselines for the new tests. We need to be certain the convection changes and regression test definition changes have no unintended effects.

File: tests/fv3_conf/hafs_fv3_run.IN [hunk 1]
```diff
@@ -25,16 +25,23 @@ elif [ ${CNTL_DIR} = "hafs_regional_1nest_atm" ] || \
...
 fi
 
-if [ ${CNTL_DIR} = "hafs_regional_atm_thompson_gfdlsf" ]; then
+if [ ${CNTL_DIR} = "hafs_regional_atm_thompson_gfdlsf" ] || [ $IMP_PHYSICS = 8 ]; then
   cp @[INPUTDATA_ROOT]/FV3_fix/qr_acr_qsV2.dat .
   cp @[INPUTDATA_ROOT]/FV3_fix/qr_acr_qgV2.dat .
   cp @[INPUTDATA_ROOT]/FV3_fix/freezeH2O.dat .
   cp @[INPUTDATA_ROOT]/FV3_fix/CCN_ACTIVATE.BIN .
 fi
+#ugwpv1
```
Comment (BinLiu-NOAA) [inline tests/fv3_conf/hafs_fv3_run.IN:41]: @SamuelTrahanNOAA, is this ugwpv1 section only for the case of  [ ${CNTL_DIR} = "gnv1_nested" ]? If so, please add it in the following if condition as well. Thanks!

File: tests/parm/input_global_hafs.nml.IN [hunk 2]
```diff
@@ -146,19 +147,29 @@
...
-       lradar         = .true.
        avg_max_length = 3600.
        h2o_phys       = .true.
        fhcyc          = @[FHCYC]
        use_ufo        = .true.
        pre_rad        = .false.
-       imp_physics    = 11
+       imp_physics    = @[IMP_PHYSICS]
+       iovr           = @[IOVR]
+       ltaerosol      = @[LTAEROSOL]
```
Comment (BinLiu-NOAA) [inline tests/parm/input_global_hafs.nml.IN:157]: Align the "=" if possible? Thanks!

File: tests/fv3_conf/hafs_fv3_run.IN (hunk 1 above)
Comment (SamuelTrahanNOAA) [reply tests/fv3_conf/hafs_fv3_run.IN:41]: The logic would be the same for any test that uses ugwpv1 regardless of its case directory.

**Comment by SamuelTrahanNOAA** [comment]
I've merged #1946 into this and tested on top of it with:
- Hera rocoto
- Jet ecflow
- Jet rocoto

Baselines:
- Jet: 
   - `/lfs4/BMC/wrfruc/Samuel.Trahan/westwater/RT/NEMSfv3gfs/develop-20231109/gnv1_c96_no_nest_debug_intel`
   - `/lfs4/BMC/wrfruc/Samuel.Trahan/westwater/RT/NEMSfv3gfs/develop-20231109/gnv1_nested_intel`
- Hera:
    - `/scratch2/BMC/wrfruc/Samuel.Trahan/westwater/RT/NEMSfv3gfs/develop-20231109/gnv1_c96_no_nest_debug_gnu`
    - `/scratch2/BMC/wrfruc/Samuel.Trahan/westwater/RT/NEMSfv3gfs/develop-20231109/gnv1_c96_no_nest_debug_intel`
    - `/scratch2/BMC/wrfruc/Samuel.Trahan/westwater/RT/NEMSfv3gfs/develop-20231109/gnv1_nested_intel`

File: tests/parm/input_global_hafs.nml.IN (hunk 2 above)
Comment (SamuelTrahanNOAA) [reply tests/parm/input_global_hafs.nml.IN:157]: Done.

File: tests/fv3_conf/hafs_fv3_run.IN (hunk 1 above)
Comment (SamuelTrahanNOAA) [reply tests/fv3_conf/hafs_fv3_run.IN:41]: I made this change.

**Review by BinLiu-NOAA — DISMISSED** [review]
Looks good to me.

File: tests/parm/cpld_control.nml.IN [hunk 3]
```diff
@@ -259,7 +262,18 @@
   knob_ugwp_dokdis   = @[KNOB_UGWP_DOKDIS]
   knob_ugwp_ndx4lh   = @[KNOB_UGWP_NDX4LH]
   knob_ugwp_version  = @[KNOB_UGWP_VERSION]
-  launch_level      = 54
+  ! This is only for UGWP v0
+@[HIDE_UGWPV0] launch_level      = 54
```
Comment (junwang-noaa) [inline tests/parm/cpld_control.nml.IN:266]: @SamuelTrahanNOAA I feel this brings the mismatched namelist issue up to the top namelist templates. The launch_level is used by both ugwpv0 and ugwpv1, it's confusing that it only shows up as a namelist list for ugwpv0 but is a hardcoded variable in UGWPV1. If ugwpv0 and ugwpv1 defines their own namelist with different variables, they should give namelist a different name. I'd suggest not bringing these discrepancy to top namelist template. Do we need to create an issue in CCPP physics?

Comment (SamuelTrahanNOAA) [reply tests/parm/cpld_control.nml.IN:266]: > Do we need to create an issue in CCPP physics?

Please open an issue. Renaming the namelists would clean things up.

For now, the new variables I've added are the best solution. It prevents duplication of nml files; that's always error-prone.

**Comment by SamuelTrahanNOAA** [comment]
I've updated all components and merged develop to this branch after #1946 was merged to develop.

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA thanks! @BrianCurtis-NOAA @zach1221 @FernandoAndrade-NOAA I think we can create and add new 
 baseline cases gnv1_c96_no_nest_debug and gnv1_nested to current baseline BL_DATE=20231108.

**Comment by SamuelTrahanNOAA** [comment]
> @SamuelTrahanNOAA thanks! @BrianCurtis-NOAA @zach1221 @FernandoAndrade-NOAA I think we can create and add new baseline cases gnv1_c96_no_nest_debug and gnv1_nested to current baseline BL_DATE=20231108.

I already updated my BL_DATE to 20231109 so it would not clash with the other baseline during my own testing. If you want the old date, you'll need to revert my bl_date.conf.

Also: I generated the baselines on Hera and Jet already. You should consider using those, to save time.

**Comment by jkbk2004** [comment]
> Member

That's fine. We can copy over to new one.

File: tests/parm/cpld_control.nml.IN (hunk 3 above)
Comment (SamuelTrahanNOAA) [reply tests/parm/cpld_control.nml.IN:266]: Jun has opened an issue, so I'm considering this conversation resolved.

Comment (jkbk2004) [reply tests/parm/cpld_control.nml.IN:266]: @mdtoyNOAA FYI

**Comment by SamuelTrahanNOAA** [comment]
The OTR and CI tests have said they're "Pending" for a while. Is something wrong?

**Comment by zach1221** [comment]
> The OTR and CI tests have said they're "Pending" for a while. Is something wrong?

No, it's running currently. No issues yet.

**Comment by jkbk2004** [comment]
@SamuelTrahanNOAA I see time out issue on /scratch1/NCEPDEV/stmp2/Jong.Kim/FV3_RT/rt_72722/gnv1_nested_intel
@zach1221 can you pick up from there?

**Comment by zach1221** [comment]
> /scratch1/NCEPDEV/stmp2/Jong.Kim/FV3_RT/rt_72722/gnv1_nested_intel

I think Jong tried increasing the wall clock. I'll try to get it to run on my end.

**Comment by zach1221** [comment]
Baseline creation on Hera for the new tests went fine, without timeouts. Running the matching now.

**Comment by SamuelTrahanNOAA** [comment]
> @SamuelTrahanNOAA I see time out issue on /scratch1/NCEPDEV/stmp2/Jong.Kim/FV3_RT/rt_72722/gnv1_nested_intel @zach1221 can you pick up from there?

It's running, but it's not finishing in time.

`
234:  THVSK_lnd(I)         181           3   360.094395904550
234:    351.458657668197        305.341506797005        305.341506797005
234:   4.043075031881894E-002
  0: PASS: fcstRUN phase 1, n_atmsteps =              180 time is         8.939060
`

From my regression test logs, it finished in four minutes:

`
  0: The total amount of wall time                        = 242.467459
`

Yours is running about ten times slower than mine which is an implausible difference. I'll rerun now to see if I can reproduce the problem. Most likely, you got a bad node and rerunning will fix it.

**Comment by SamuelTrahanNOAA** [comment]
I've run the gnv1_nested_intel test under my account five times, and all of them finished in under five minutes.

Edit: Those tests were on Hera, the machine where another user's gnv1_nested_intel hit its wallclock limit.

**Comment by FernandoAndrade-NOAA** [comment]
Ok, testing is complete. This should be ready to continue the merge process, I'll leave a note in the sub PRs.

**Comment by SamuelTrahanNOAA** [comment]
The FV3 submodule points to the head of develop and I've reverted .gitmodules. This PR is ready for final review and merge.

**Review by sadeghitabas — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Pull request merged**